### ISSUE TITLE
Common engine

### ISSUE THREAD

**Issue by ultraviolet-jordan** [LostCityRS/Engine#8]
Why is a brand new project being coded when there can just be a common engine created which can just be extracted from the 2004scape project and this engine works for all revisions? (it's a trick question). massive waste of time for no reason.

**Comment by ultraviolet-jordan** [comment]
all of the refactoring on 2004scape was done with this concept in mind already. fuck me i guess.

**Comment by ultraviolet-jordan** [comment]
How to create an osrs server in 3 hours:
- Take 2004scape project
- Add js5
- Implement packets under its own folder.
- Maybe add 1 env feature flag.
- Done.

**Comment by Pazaz** [comment]
This project is a ground-up rethinking of some ideas and it isn't a waste of time to do so.
Opening these types of issues unfortunately just clutters the history of the repo, they are not constructive.

Previous work isn't being discarded, but even if it was, that's okay. This is just a hobby for people to spend a few hours on after work anyways.
You're looking at a small playground I use to experiment and iterate on our lessons learned, AND give Zeruth/anyone else a place to commit the network encoders and decoders for the client refactor. *Perhaps your issue is with the name - I didn't have anything better to call it.*

We haven't written perfect engine code in 2004. Some stuff was over engineered or originally designed with different information, and adapted to fit knowledge gained over time. Yes I'm guilty of writing that code and still work to improve/rewrite it today.

It is not a *3 hour job*, but let's say it is. Could I implore you to spend an evening?
Here's a few problems you'll run into:
- 2004Scape has a hard reliance on the cache being broken out and packed fully
- 2004Scape has hardcoded 225 code for connection negotiation (Login is not abstracted with encoder/decoders)
- 2004Scape's interface system needs rewriting to have the redesigned ID space and if3 in mind
- bug fix: 2004's weak queues are clearing too frequently, some rethinking needed

You can say none of these add up to starting a new project. It's still been a huge help to test and redesign individual components, rather than a big push to make something completely functional all at once. And it's been fun.

---

Semi-related: I've been copying over network improvements figured out here **back into 2004Scape**. The eventual goal is we will have a single engine, with code either copied 1:1 or improved on.

**Comment by ultraviolet-jordan** [comment]
"You can say none of these add up to starting a new project"
None of these add up to starting a new project.

I'm not spending an evening on anything, the server *is* finished already, excluding some members content.
If there's a bug, then make an issue, and someone fix it. Why is there an entire new project taking priority over that??

There's always an (new) excuse isn't there bro.... Stop gatekeeping 2004scape so that we can all finally play the game we spent endless hours working on......

I'm not here to be constructive, I'm here to be disappointed because my time, effort, and energy, with the expectation of the server going live, is not being met.

**Comment by Pazaz** [comment]
Thanks for the discussion, but I think you're on the wrong forum then. My DMs are open if you have sincere problems.

A couple people showed interest in contributing time to an idea, I cleaned up a client to get them started and created this repo to connect to it. I'm not putting 100% of my effort here and I think your aggressiveness is unwarranted. I spent some hobby time to take a refreshed look at some old problems; my cumulative time spent has been 2 or 3 days, and I haven't worked on it in weeks.

I could've called this repo Playground, ResearchEngine, or OSServer, but I picked Engine to keep it simple. It just needed to exist for people to push to.

I understand you want to see the other project go live and users playing with everyone's work, and you dedicated time to help make that a reality. It'll happen.
